libs/template/if/src/lib/if.directive.ts

```diff
@@ -486,7 +491,20 @@ export class RxIf<T = unknown>
     return this.then ? this.then : this.templateRef;
   }
 
-  constructor(private readonly templateRef: TemplateRef<RxIfViewContext<T>>) {}
+  /**
+   * We will store a signal inside a signal in order to listen to both value changes, and reference changes.
+   * @internal
+   */
+  private currentRxIfSignal = signal<Signal<T> | undefined>(undefined);
+
+  constructor(private readonly templateRef: TemplateRef<RxIfViewContext<T>>) {
+    effect(() => {
+      if (this.currentRxIfSignal() !== undefined) {
+        const rxIfValue = this.currentRxIfSignal()();
+        untracked(() => this.templateNotifier.next(rxIfValue));
+      }
+    });
+  }
```

Comment:
isn't rxIfValue of type `Signal` in this case? Shouldn't it be `currentRxIfSignal()()?`

Reply:
That's how it is already 🤔

Reply:
lol, I'm blind